lib_staging.ks

```diff
@@ -0,0 +1,63 @@
+global stagingFuelList	is list("LiquidFuel", "SolidFuel").
+global stagingNumber	is -1.
+global stagingEngines	is list().
+global stagingTanks		is list().
+
+// to be called whenever current stage changes to prepare data for quicker test
+function stagingPrepare {
+	wait until stage:ready.
+	set stagingNumber	to stage:number.
+	set stagingEngines	to list().
+	set stagingTanks	to list().
+	if stage:number = 0 return.
+
+//	prepare list of engines that are in current stage but not in next stage
+	list engines in engines.
+	for e in engines if e:stage = stage:number and e:parent:stage = stage:number-1
+		stagingEngines:add(e).
+
```

Comment:
While most parts have their stage number reflect the stage they separate in, engines have this number based on stage they are activated in. That is the reason I use e:parent:stage, which works for any engine that is not attached directly to another engine. Guess which engines can - boosters, but that means booster attached directly to another booster and that means that the last booster in the chain (attached to decoupler) will be matched and that is enough for it to work (assuming all the boosters attached to each other will have same limits - it would fail if the core booster is depleted earlier than the other attached to it... which does not sound like a reasonable design). 

Boosters are also tanks and active boosters won't be matched as tanks, because `t:stage = stage:number-1` will fail for them (they have `t:stage = stage:number`), which is neither good nor bad, they will be staged as engines. **BUT!** What about boosters activated in next stage? They could pontentially block tank staging. I will have to test that.

Maybe we can just remove 'SolidFuel' from the list, that should fix it. Why having it there? But I would rather keep it as a list that you can add Karbonite and what not.

Reply:
Agree; merged your latest changes to my working branch.

As for code style and contribution: nice of you to think of it, but delegates make code much more modular so I'll merge with `lib_staging` as a separate thing.

I've had some more ideas about making RAMP modular and space-efficient; I'll add another issue to discuss. I think a `lib/` dir is a good idea, and I think you're already using that on your fork?